lib/compress/zstd_compress.c

```diff
@@ -2372,9 +2381,21 @@ static size_t ZSTD_compress_frameChunk (ZSTD_CCtx* cctx,
                 cSize = ZSTD_noCompressBlock(op, dstCapacity, ip, blockSize, lastBlock);
                 FORWARD_IF_ERROR(cSize);
             } else {
-                U32 const cBlockHeader24 = lastBlock + (((U32)bt_compressed)<<1) + (U32)(cSize << 3);
-                MEM_writeLE24(op, cBlockHeader24);
-                cSize += ZSTD_blockHeaderSize;
+                U32 cBlockHeader;
```

Comment:
This needs to be done inside of `ZSTD_compressBlock_internal()`, because when we use an RLE block we need to not swap `prevCBlock` and `nextCBlock` https://github.com/facebook/zstd/blob/dev/lib/compress/zstd_compress.c#L2296.

Then we'll ned some way to signal if the block is RLE or not. Returning `cSize == 1` for RLE would work.